src/App.tsx

```diff
@@ -1,223 +1,160 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useCallback,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { getTodos, addTodo, removeTodo } from './api/todos';
+import { Errors } from './components/Errors';
+import { Footer } from './components/Footer';
+import { FilterMethods } from './types/FilterMethods';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterMethod, setFilterMethod]
+  = useState<FilterMethods>(FilterMethods.ALL);
+  const [title, setTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+
+  const getTodosFromsServer = useCallback(async () => {
+    if (!user) {
+      return;
+    }
+
+    try {
+      const todosFromServer = await getTodos(user.id);
+
+      setTodos(todosFromServer);
+    } catch (Error) {
+      setErrorMessage('Loading error!');
+
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, []);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
-  }, []);
+
+    getTodosFromsServer();
+  }, [todos]);
+
+  const filteredTodos = todos.filter(({ completed }) => {
+    switch (filterMethod) {
+      case FilterMethods.COMPLETED:
+        return completed;
+
+      case FilterMethods.ACTIVE:
+        return !completed;
+
+      default:
+        return true;
+    }
+  });
+
+  const addNewTodo = async (newTitle: string) => {
+    if (user) {
+      try {
+        setIsAdding(true);
+
+        const todoToAdd = {
+          title: newTitle,
+          userId: user.id,
+          completed: false,
+        };
+
+        const newTodo = await addTodo(todoToAdd);
+
+        setIsAdding(false);
+
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+      } catch (error) {
+        setErrorMessage('Unable to add a todo');
+        setTitle('');
+        setIsAdding(false);
+
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      }
+    }
```

Comment:
Let's avoid unnecessary nesting.
```suggestion
      if (!user) return;
      try {
        setIsAdding(true);

        const todoToAdd = {
          title: newTitle,
          userId: user.id,
          completed: false,
        };

        const newTodo = await addTodo(todoToAdd);

        setIsAdding(false);

        setTodos(currentTodos => [...currentTodos, newTodo]);
      } catch (error) {
        setErrorMessage('Unable to add a todo');
        setTitle('');
        setIsAdding(false);

        setTimeout(() => {
          setErrorMessage('');
        }, 3000);
      }
    
```